dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java

```diff
@@ -86,15 +91,17 @@ public Object onCompleted(Response response) {
             successHandler(response);
         } else {
             postResponseTelemetry.validateResponseCode(meterStatsManager, statusCode);
-            failureHandler("Received status code : " + statusCode);
+            boolean shouldFailOnError = (httpSourceConfig.isFailOnErrors() ? shouldFailOnError(statusCode) : false);
+            failureHandler("Received status code : " + statusCode, shouldFailOnError);
```

Comment:
i would suggest to keep the method signature same. 
and in line 131 do 

if( shouldFailOnError(statusCode)) {
...
}

then the method sholdFailOnError will do the necessary checks

Reply:
Made changes